### DISCUSSION TITLE
Migrating help-- screen flashes

### DISCUSSION THREAD

**Discussion by keijiota** [jspsych/jsPsych#3023] [Q&A]
Hello jsPsych community! 

I have got a migration issue from version 6 to version 7. The image files I use sometimes disappear from the screen at the moment when one timeline transits to the next timeline. The image files are supposed to be always there on the screen and there was no blank on version 6. Please find a video record attached below. 

https://user-images.githubusercontent.com/36744375/203094390-fc5de6c4-2939-4684-b57f-1b1e22d43544.mov

It's like a screen-flipping issue and not good for eyes. This happens occasionally and not in all trials. Could I ask what is the problem and how I can avoid them? It would be appreciated if someone could help with this. 

Many thanks!
Keiji

**Comment by jodeleeuw** [comment]
Hi @keijiota 

What plugins are you using for these trials? Can you share a snippet of the timeline?

**Comment by keijiota** [comment]
Hi @jodeleeuw 
Thank you for your reply! I use plugins "jsPsychPsychophysics" which is developed by Kuroki (https://jspsychophysics.hes.kyushu-u.ac.jp/) 
Two variables below "untilkeypress" and "afterkeypress" are presented in order on the timeline
![image](https://user-images.githubusercontent.com/36744375/203110867-0461894f-6c45-46fc-8f25-a64a36a45814.png)

![image](https://user-images.githubusercontent.com/36744375/203111781-a086cf07-24c9-44a5-b601-f336dcb22b0f.png)

**Comment by nikbpetrov** [comment]
I would check the obvious things first. Given that it happens only on a selection of the trials, it's possible that it _technically_ happens on all trials but you only perceive it on some, most likely due to screen refresh rate (perhaps on some trials the event loop is executed towards the end of the screen refresh rate duration - or whatever it was - such that is imperceptible). The change from v6 to v7 could be due to some minor additional load associated with v7 or change it when functions are executed. You also have functions that need to be run for the prompt and onfinish, all of which provide additional overhead.

This might turn out to be reasonable, because, technically, you would expect at least some gap between trials, where the contents of the previous trials are removed and the contents of the new one are shown - therefore your gap. If I am correct, this is relatively easy to check by simply adding some heavy function in your `on_finish` function on the `untilkeypress` trial - e.g. log something 100_000 times. You have plenty of overhead in the `on_finish` function in addition to your timeline looping (a function that also needs to be evaluated), and there goes your flash.

**Comment by kurokida** [comment]
Hi, I am a developer of the plugin @keijiota used.
I received the program from them, and the disappearance (flash) occured at "all" timeline changes at least in my PC.

Their program seems to use two timelines to present a series of a moving stimulus. The stimulus always disappeared at the time from one to the other timeline. The core jsPsych probably refresh the screen every after a timeline finishes. I belive this change was made when upgrading from V6 to V7 of jsPsych.

My suggsetion is that they should use a single timeline to present the stimulus if possible.

**Comment by keijiota** [comment]
Thank you, everyone! 
The reason why I try to upgrade the version is that I would like to use the plugin "sketchpad". I am ok with sticking to version 6 if this is available. Would there be any chance for you, @jodeleeuw, to develop the sketchpad for version 6?
